opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp

```diff
@@ -227,7 +227,19 @@ namespace Opm {
 
         using namespace std::placeholders;
         const auto handler = std::bind( handleKW, _1, schedule, props, n_xyz );
+
+#ifdef _MSC_VER
+        for (auto& x : section)
+        {
+            auto keywords = handler(x);
+            for (auto& keyword : keywords)
+            {
+                this->keywords.push_back(keyword);
+            }
+        }
+#else
         this->keywords = fun::concat( fun::map( handler, section ) );
```

Comment:
> but it is still better than having both codepaths and some preprocessor-foo? I personally prefer some minor duplication of trivial code if it makes the code simpler in the sense that it reduces the mental load of stuff which you need to know in order to understand it. (remember the recent 'leftpad' incident?) That's my personal opinion, though.

The preprocessor solution is unacceptable, yes. But the solution should then go into `fun` or similar considerations, not abandoning all kind of abstraction. Leftpad has nothing to do with this and abstractions **reduce** the mental load. If it didn't we'd all still be using vacuum tubes, just to go on a tangent.

> most places where these are used are in no way performance critical, I suppose?

It's not just for efficiency, it's for generality and applicability. Want to populate a vector? Fine. Want to populate a list? Fine, same code. Want to populate a `std::map`? No problem. Want to not populate anything, but rather evalute-and-reduce? Got your back.

> if we were talking lisp or scheme, I'd agree. But in C++, these are not standard constructs, and people are not used to the functional programming paradigm...

This is just not true.